src/components/app.tsx

```diff
@@ -244,45 +294,48 @@ export const App: React.FC<IAppProps<IModelInputState, IModelOutputState, IModel
-            isSaved={isSaved}
-            isFinished={isFinished}
-            onSaveClicked={saveToSelectedContainer}
-            currentTimeLabel={t(timeLabel, {vars: {weeks: `${weeks}`}})}
-            currentTime={time}
-            maxTime={1}
-            savedBgColor={kSavedBgColor}
-          >
-            <SimulationView
-              backgroundImage={backgroundImage}
-              environment={environment}
-              leafPackState={leafPackState}
-              fish={fish}
-              onShowTray={() => setShowTray(true)}
+        <DndProvider backend={TouchBackend} options={{enableMouseEvents: true}} >
+          <div className="simulation" data-testid="simulation">
+            <MainViewWrapper
+              title={selectedContainerId}
+              isSaved={isSaved}
               isFinished={isFinished}
-              isRunning={isRunning}
-            />
-            <Tray
-              trayAnimals={trayAnimals}
-              onHideTray={() => setShowTray(false)}
-              onTrayObjectSelect={handleTrayObjectSelect}
+              onSaveClicked={saveToSelectedContainer}
+              currentTimeLabel={t(timeLabel, {vars: {weeks: `${weeks}`}})}
+              currentTime={time}
+              maxTime={1}
+              savedBgColor={kSavedBgColor}
+            >
+              <SimulationView
+                backgroundImage={backgroundImage}
+                environment={environment}
+                leafPackState={leafPackState}
+                fish={fish}
+                onShowTray={() => setShowTray(true)}
+                isFinished={isFinished}
+                isRunning={isRunning}
+              />
+              <Tray
+                trayObjects={trayObjects}
+                onHideTray={() => setShowTray(false)}
+                onTrayObjectSelect={handleTrayObjectSelect}
+                traySelectionType={traySelectionType}
+                hidden={!showTray}
+                onTrayObjectMove={handleMoveTrayObject}
+                isRunning={isRunning}
+              />
+            </MainViewWrapper>
+            <Notebook
+              trayObjects={trayObjects}
+              environment={environment}
+              featureSelections={habitatSelectedFeatures}
+              onSelectFeature={handleHabitatSelectFeature}
+              onCategorizeAnimal={handleCategorizeAnimal}
               traySelectionType={traySelectionType}
-              hidden={!showTray}
               isRunning={isRunning}
             />
-          </MainViewWrapper>
-          <Notebook
-            trayAnimals={trayAnimals}
-            environment={environment}
-            featureSelections={habitatSelectedFeatures}
-            onSelectFeature={handleHabitatSelectFeature}
-            onCategorizeAnimal={handleCategorizeAnimal}
-            traySelectionType={traySelectionType}
-            isRunning={isRunning}
-          />
-        </div>
+          </div>
+        </DndProvider>
```

Comment:
Not necessarily for this PR, but given the growing length and complexity of `app.tsx`, it would be worth thinking about whether there are any parts of it that could reasonably be split off into their own module/component.

Reply:
Fixed.